exonum/src/runtime/types.rs

```diff
@@ -367,59 +367,94 @@ impl Display for ArtifactStatus {
+            }
         })
     }
 }
 
+// TODO Investigate boilerplate-less approach of enums usage as binary values and keys. [ECR-3941]
+
 /// Status of a service instance.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
 pub enum InstanceStatus {
-    /// The service instance is pending deployment.
-    Pending = 0,
-    /// The service instance has been successfully deployed.
+    /// The service instance is active.
     Active = 1,
+    /// The service instance is stopped.
+    Stopped = 2,
+}
+
+impl InstanceStatus {
+    /// Indicates whether the service instance status is active.
+    pub fn is_active(self) -> bool {
+        self == InstanceStatus::Active
+    }
 }
 
 impl Display for InstanceStatus {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match self {
             InstanceStatus::Active => f.write_str("active"),
-            InstanceStatus::Pending => f.write_str("pending"),
+            InstanceStatus::Stopped => f.write_str("stopped"),
         }
     }
 }
 
-impl ProtobufConvert for InstanceStatus {
-    type ProtoStruct = schema::runtime::InstanceStatus;
-
-    fn to_pb(&self) -> Self::ProtoStruct {
-        match self {
-            InstanceStatus::Active => schema::runtime::InstanceStatus::SERVICE_ACTIVE,
-            InstanceStatus::Pending => schema::runtime::InstanceStatus::SERVICE_PENDING,
+impl InstanceStatus {
+    // This method must have an exact signature so that is can be used with the
+    // `protobuf_convert(with)`.
+    #[allow(clippy::trivially_copy_pass_by_ref, clippy::wrong_self_convention)]
+    fn to_pb(status: &Option<InstanceStatus>) -> schema::runtime::InstanceState_Status {
+        match status {
+            None => schema::runtime::InstanceState_Status::NONE,
+            Some(InstanceStatus::Active) => schema::runtime::InstanceState_Status::ACTIVE,
+            Some(InstanceStatus::Stopped) => schema::runtime::InstanceState_Status::STOPPED,
         }
     }
 
-    fn from_pb(pb: Self::ProtoStruct) -> Result<Self, failure::Error> {
+    fn from_pb(
+        pb: schema::runtime::InstanceState_Status,
+    ) -> Result<Option<InstanceStatus>, failure::Error> {
         Ok(match pb {
-            schema::runtime::InstanceStatus::SERVICE_ACTIVE => InstanceStatus::Active,
-            schema::runtime::InstanceStatus::SERVICE_PENDING => InstanceStatus::Pending,
+            schema::runtime::InstanceState_Status::NONE => None,
+            schema::runtime::InstanceState_Status::ACTIVE => Some(InstanceStatus::Active),
+            schema::runtime::InstanceState_Status::STOPPED => Some(InstanceStatus::Stopped),
         })
     }
 }
 
+impl BinaryValue for InstanceStatus {
+    fn to_bytes(&self) -> Vec<u8> {
+        (*self as u32).to_bytes()
+    }
+
+    fn from_bytes(bytes: Cow<'_, [u8]>) -> Result<Self, failure::Error> {
+        let code = u32::from_bytes(bytes)?;
+        match code {
+            1 => Ok(InstanceStatus::Active),
```

Comment:
Nit: Isn't it possible to use `InstanceState_Status` constants here? Also, what's stopping from using `from_pb` / `to_pb` methods in `BinaryValue` implementation? enum variants seem to be encoded with the LEB128 encoding, so the current `BinaryValue` impl is not consistent with the Protobuf presentation.

Reply:
It is possible but painful since `InstanceState_Status` has no `repr(i32)` attribute.
Also this type doesn't implement `rust_protobuf::Message` trait, so I can't just use `ProtobufConvert`.

I think that the best solution is extending `BinaryValue` macros to support enums with discriminant.
ECR-3941